### PR TITLE
Update generate_binaries.jl

### PR DESCRIPTION
**Pull request by amontoison**
Copy the license of each dependency when we generate the precompiled archives.

### CODE REVIEW

**Review by cvanaret — APPROVED** [review]

**Comment by amontoison** [comment]
@cvanaret 
Can you temporary comment these lines in `generate_binaries.jl` ?
```
      for folder in readdir("products/$platform/deps/licenses")
        cp("products/$platform/deps/licenses/$folder", "products/$platform/share/licenses/$folder")
      end
      rm("products/$platform/deps/licenses", recursive=true)
 ```
I don't why but I have an error on my fork and the CI build for a new release will fail because of that.

**Comment by cvanaret** [comment]
> @cvanaret Can you temporary comment these lines in `generate_binaries.jl` ?

Addressed here: https://github.com/cvanaret/Uno/pull/99
I retagged v1.3.0 :crossed_fingers:

**Comment by cvanaret** [comment]
I still get the error with `rm -rf /opt/${target}/${target}/sys-root/System`.

**Comment by amontoison** [comment]
I don't have this issue with a a local build.
So I don't why we have that with GHA.

**Comment by cvanaret** [comment]
Could it be a matter of permissions?

File: .github/julia/generate_binaries.jl [hunk 1]
```diff
@@ -47,6 +30,12 @@ for (platform, libdir, ext) in platforms
       # Unzip the tarball of the dependencies
       run(`tar -xzf products/$platform/deps.tar.gz -C products/$platform`)
 
+      # Copy the license of each dependency
+      for folder in readdir("products/$platform/deps/licenses")
+        cp("products/$platform/deps/licenses/$folder", "products/$platform/share/licenses/$folder")
```
Comment (cvanaret) [inline .github/julia/generate_binaries.jl:35]: Do we need
`mkdir "products/$platform/share`
`mkdir "products/$platform/share/licenses`
here?

**Comment by cvanaret** [comment]
The binary generation failed with:
```shell
Run julia --color=yes .github/julia/generate_binaries.jl
ERROR: LoadError: IOError: mkdir("products/aarch64-apple-darwin-cxx11/share/licenses/ASL"; mode=0o777): no such file or directory (ENOENT)
```
see https://github.com/cvanaret/Uno/actions/runs/11879298357/job/33101034439#step:9:10

**Pull request merged**